client/js/upload.js

```diff
@@ -0,0 +1,53 @@
+"use strict";
+const $ = require("jquery");
+const socket = require("./socket");
+const SocketIOFileUpload = require("socketio-file-upload/client");
+const instance = new SocketIOFileUpload(socket);
+
+instance.listenOnInput(document.getElementById("upload-input"));
+instance.listenOnDrop(document.getElementById("chat-container"));
+
+$("#upload").on("click", () => {
+	$("#upload-input").trigger("click");
+});
+
+instance.addEventListener("complete", function() {
+	// Reset progressbar
+	$("#upload-progressbar").width(0);
+});
+
+instance.addEventListener("progress", function(event) {
+	const percent = `${((event.bytesLoaded / event.file.size) * 100)}%`;
+	$("#upload-progressbar").width(percent);
+});
+
+instance.addEventListener("error", function(event) {
+	// Reset progressbar
+	$("#upload-progressbar").width(0);
+	$("#loading-page-message, #connection-error").addClass("shown");
+	$("#loading-page-message, #connection-error").text(event.message);
```

Comment:
Chain `text` after `addClass`. You also don't need to target `#loading-page-message`.

Reply:
I know I don't need to, but its targeted elsewhere so did this for consistency

Reply:
I know what you mean, but in other places it makes sense because the error can be shown before the app fully loads.

In this case uploads only work after the app loaded, so there is no loading page.